coremltools/models/_deprecation.py

```diff
@@ -0,0 +1,26 @@
+import warnings
+import functools
+
+def deprecated(obj=None, str=""):
+    '''
+    Decorator to mark a function or a class as deprecated
+    '''
+
+    def decorator_deprecation_warning(obj):
+        @functools.wraps(obj)
+        def wrapped(*args, **kwargs):
+            if isinstance(obj, type):
+                msg = "Class \"%s\" is deprecated and will be removed in the next release" % obj.__name__
+            else:
+                msg = "Function \"%s\" is deprecated and will be removed in the next release" % obj.__name__
+            if str:
```

Comment:
Awesome. 

Just a last minor comment. `str` is a builtin-in python keyword. Here we are overriding the definition but in general `if str` always evaluates to True.

Reply:
Good point. Changed it.